credential/util.go

```diff
@@ -33,8 +33,13 @@ func ToCredential(genericCred any) (jws.Headers, jwt.Token, *VerifiableCredentia
 		verifiableCredential := genericCred.(VerifiableCredential)
 		return nil, nil, &verifiableCredential, nil
 	case string:
-		// TODO(gabe) support the case where the string is a json representation of an LD credential
-		// JWT
+		// first try the case where the string is JSON of a VC object
+		var cred VerifiableCredential
+		if err := json.Unmarshal([]byte(genericCred.(string)), &cred); err == nil {
```

Comment:
instead of casting here, you can use the object from the swith

```go
switch typedCred := genericCred.(type) {
  // ...
  case string:
    var cred VerifiableCredential
    if err := json.Unmarshal([]byte(typedCred), &cred); err == nil {
    // ...
}
```

Reply:
thanks updated here and elsewhere